clippy_lints/src/methods/mod.rs

```diff
@@ -1783,6 +1784,33 @@ declare_clippy_lint! {
     "using `as_ref().map(Deref::deref)`, which is more succinctly expressed as `as_deref()`"
 }
 
+declare_clippy_lint! {
+    /// ### What it does
+    /// Checks for usage of `_.as_ref().map(Deref::deref)` or its aliases
+    /// (such as `String::as_str`) on `Result`.
+    ///
+    /// ### Why is this bad?
+    /// Readability, this can be written more concisely as
+    /// `_.as_deref()`.
+    ///
+    /// ### Example
+    /// ```no_run
+    /// # let res = Ok::<_, ()>("".to_string());
+    /// res.as_ref().map(String::as_str)
+    /// # ;
+    /// ```
+    /// Use instead:
+    /// ```no_run
+    /// # let res = Ok::<_, ()>("".to_string());
+    /// res.as_deref()
+    /// # ;
+    /// ```
+    #[clippy::version = "1.83.0"]
```

Comment:
nit: you should update this for 1.84.0 (due to the review delay, not your fault)